api/src/main/java/com/flipkart/poseidon/internal/ListDataType.java

```diff
@@ -24,7 +24,7 @@
 import java.util.Collection;
 import java.util.List;
 
-public class ListDataType<T> extends ArrayList<T> implements DataType {
+public class ListDataType<T extends DataType> extends ArrayList<T> implements DataType {
```

Comment:
You may want to look at OrchestratorDataSource.java. ListDataType and MapDataType are used to convert response coming out of Hydra's composer to a DataType (to maintain DataSource contract of returning DataType). ListDataType is not used to convey a list of datatypes :)